### PR TITLE
web view win32

### PR DESCRIPTION
**Pull request by summerinsects**

### CODE REVIEW

**Comment by summerinsects** [comment]
1.add simple implement of WebView for win32
2.Modify `std::string' to`const std::string &' as issues 7958 said
3.Open the test case in cpp-tests
4.Modify spelling mistake

**Comment by zilongshanren** [comment]
@wohaaitinciu 
Thanks, could you please fix the compile error on win32 platform?

**Comment by summerinsects** [comment]
@andyque 
I'll try my best.
But I don't know what' wrong with the linking error

**Comment by minggo** [comment]
The error is 

```
>d:\jenkins_slave\workspace\cocos-2dx-pull-request-build\node\win32_win7\cocos\ui\UIWebView-inl.h(83): error C2039: 'loadURL' : is not a member of 'cocos2d::experimental::ui::WebViewImpl' (..\ui\UIWebView.cpp)
3>          d:\jenkins_slave\workspace\cocos-2dx-pull-request-build\node\win32_win7\cocos\ui\UIWebViewImpl_win32.h(30) : see declaration of 'cocos2d::experimental::ui::WebViewImpl'
3>  CCActionGrid.cpp
3>  CCActionGrid3D.cpp
3>D:\Jenkins_slave\workspace\cocos-2dx-pull-request-build\node\win32_win7\cocos\ui/UIHelper.h : warning C4819: The file contains a character that cannot be represented in the current code page (936). Save the file in Unicode format to prevent data loss (..\ui\UIWidget.cpp)
3>..\ui\UIWebViewImpl_win32.cpp(107): error C2039: 'shouldStartLoading' : is not a member of 'cocos2d::experimental::ui::WebView'
3>          d:\jenkins_slave\workspace\cocos-2dx-pull-request-build\node\win32_win7\cocos\ui\UIWebView.h(42) : see declaration of 'cocos2d::experimental::ui::WebView'
3>..\ui\UIWebViewImpl_win32.cpp(109): error C2039: 'shouldStartLoading' : is not a member of 'cocos2d::experimental::ui::WebView'
3>          d:\jenkins_slave\workspace\cocos-2dx-pull-request-build\node\win32_win7\cocos\ui\UIWebView.h(42) : see declaration of 'cocos2d::experimental::ui::WebView'
3>..\ui\UIWebViewImpl_win32.cpp(114): error C2039: 'didFinishLoading' : is not a member of 'cocos2d::experimental::ui::WebView'
3>          d:\jenkins_slave\workspace\cocos-2dx-pull-request-build\node\win32_win7\cocos\ui\UIWebView.h(42) : see declaration of 'cocos2d::experimental::ui::WebView'
3>..\ui\UIWebViewImpl_win32.cpp(116): error C2039: 'didFinishLoading' : is not a member of 'cocos2d::experimental::ui::WebView'
3>          d:\jenkins_slave\workspace\cocos-2dx-pull-request-build\node\win32_win7\cocos\ui\UIWebView.h(42) : see declaration of 'cocos2d::experimental::ui::WebView'
```

**Comment by minggo** [comment]
I think it is because you should use correct namespace `cocos2d::experimental`.

**Comment by summerinsects** [comment]
Oh, I see. The callback functions has been renamed

**Comment by summerinsects** [comment]
@andyque 
@minggo 
It takes me very long time to fix the linking error
please review my code

**Comment by zilongshanren** [comment]
@wohaaitinciu 
Thanks very much. I really appreciate your hard work.
Hi, @minggo  could we merge it after release v3.3 beta?

File: cocos/2d/libcocos2d.vcxproj [hunk 1]
```diff
@@ -429,6 +429,8 @@ xcopy /Y /Q "$(ProjectDir)..\..\external\chipmunk\prebuilt\win32\release-lib\*.*
     <ClCompile Include="..\ui\UITextBMFont.cpp" />
     <ClCompile Include="..\ui\UITextField.cpp" />
     <ClCompile Include="..\ui\UIVBox.cpp" />
+    <ClCompile Include="..\ui\UIWebView.cpp" />
+    <ClCompile Include="..\ui\UIWebViewImpl_win32.cpp" />
```
Comment (zilongshanren) [inline cocos/2d/libcocos2d.vcxproj]: please use `UIWebViewImpl-win32.cpp

File: cocos/2d/libcocos2d.vcxproj [hunk 2]
```diff
@@ -807,6 +809,9 @@ xcopy /Y /Q "$(ProjectDir)..\..\external\chipmunk\prebuilt\win32\release-lib\*.*
     <ClInclude Include="..\ui\UITextBMFont.h" />
     <ClInclude Include="..\ui\UITextField.h" />
     <ClInclude Include="..\ui\UIVBox.h" />
+    <ClInclude Include="..\ui\UIWebView-inl.h" />
+    <ClInclude Include="..\ui\UIWebView.h" />
+    <ClInclude Include="..\ui\UIWebViewImpl_win32.h" />
```
Comment (zilongshanren) [inline cocos/2d/libcocos2d.vcxproj]: The same, use hyphen instead of underscore to name file.

**Comment by summerinsects** [comment]
Should I copy the Copyright Information from other files?

**Comment by minggo** [comment]
Yep. Please, and you can add your personal information there.

**Comment by minggo** [comment]
@andyque 
Yep, i will merge it after releasing v3.3beta0.

**Comment by zilongshanren** [comment]
@wohaaitinciu 
Hi, I have tested your Win32 WebView implementation. 
Here is my feedback:
1. The TextField seems can't be touched any more. 
2. "Forward" and "back" is not working.
3. I can't load HTML page.

**Comment by zilongshanren** [comment]
@wohaaitinciu 
Sorry, it's my fault. 
Everything works now.
It is really awesome, thanks again for your hard work.:)

**Comment by summerinsects** [comment]
My pleasure :)

**Pull request merged**